tests/cmake/gen_rst.cmake

```diff
@@ -42,10 +41,13 @@ endmacro()
 ct_add_test(NAME test_gen_rst)
 function(${test_gen_rst})
 
+  #TODO: This is a hack
+  set(config_file "${cwd}/../../src/cminx/config_default.yaml")
+
   ct_add_section(NAME test_document)
   function(${test_document})
     do_setup()
-    cminx_gen_rst("${input_file}" "${output_dir}")
+    cminx_gen_rst("${input_file}" "${output_dir}" "-s" "${config_file}")
```

Comment:
`config_default.yml` should be automatically loaded by the python code using the cminx module location to find it. If it's unable to locate it then we should probably fix it in the code rather than working around it.

Reply:
AFAIK cminx finds it fine when you do a pip install, the problem is for the PyInstaller install. In that case it doesn't seem to find the `config_default.yml` file next to it. I'll play around with it some more and see if I can work something out.

Reply:
Okay I think I worked out a solution for this now too. We now install the default config file to `etc/cminx`. Then in CMinx's `main` function, we check if CMinx is running as an executable. If it is, we then load the settings from `etc/cminx`. This is done first thing so that user specified settings can override the default settings.

Reply:
I think there might be a distinction in the Confuse library between the config_default file and other files, but I'll have to look at it later. I also know by default it automatically searches /etc/cminx for a config.yaml anyway, along with a few other standard config directories

Reply:
The problem with relying on the default /etc/cminx path is that there's no guarantee that CMinx is installed to system root. For example, during the testing phase it's installed to `${CMAKE_BINARY_DIR}/stage` (similar scenario will happen when CMakeTest pulls CMinx in via FetchContent)

Reply:
I tried adding `--add-data /full/path/to/config_default.yaml` to the PyInstaller call, but the resulting executable still doesn't seem to be able to find th config file automatically.

Reply:
I found [this](https://github.com/pyinstaller/pyinstaller/issues/1229) but the links seem to be dead.

Reply:
FWIW, I verified that with the `--add-data` command, PyInstaller puts the `config_default.yaml` file in the temporary directory. So this is presumably a problem with how `confuse` looks for that file. 

So I think we have two options:
1. Stick with the `etc` solution I added
2. Explicitly instruct the executable to look for the file in the temp directory.

Preference? I think 2 is simpler and more self-contained. So I'm leaning that way.

Reply:
> I tried adding `--add-data /full/path/to/config_default.yaml:.` to the PyInstaller call, but the resulting executable still doesn't seem to be able to find th config file automatically.

Try replacing the `.` destination with cminx, cminx expects the file to be in the same folder as the top level __init__.py, which would be under cminx.

If that doesn't work then I agree that we should try option 2

Reply:
@AutonomicPerfectionist that seemed to work locally. Thanks!!!!